src/hardware/serialport/nullmodem.cpp

```diff
@@ -25,96 +25,90 @@
 #include "serialport.h"
 #include "nullmodem.h"
 
-CNullModem::CNullModem(Bitu id, CommandLine* cmd):CSerial (id, cmd) {
-	Bitu temptcpport=23;
-	memset(&telClient, 0, sizeof(telClient));
-	InstallationSuccessful = false;
-	serversocket = 0;
-	clientsocket = 0;
-	serverport = 0;
-	clientport = 0;
-
-	rx_retry = 0;
+CNullModem::CNullModem(const uint8_t port_idx, CommandLine *cmd)
+        : CSerial(port_idx, cmd),
+          telClient({})
+{
+	uint32_t temptcpport = 23;
 	rx_retry_max = 20;
 	rx_state=N_RX_DISC;
 
 	tx_gather = 12;
-	
-	dtrrespect=false;
-	tx_block=false;
-	receiveblock=false;
-	transparent=false;
-	telnet=false;
-	
-	Bitu bool_temp=0;
+	uint32_t bool_temp = 0;
 
 	// usedtr: The nullmodem will
 	// 1) when it is client connect to the server not immediately but
 	//    as soon as a modem-aware application is started (DTR is switched on).
 	// 2) only receive data when DTR is on.
-	if (getBituSubstring("usedtr:", &bool_temp, cmd)) {
+	if (getUintFromString("usedtr:", bool_temp, cmd)) {
 		if (bool_temp==1) {
 			dtrrespect=true;
 			transparent=true;
 			DTR_delta=false; // connect immediately when DTR is already 1
 		}
 	}
 	// transparent: don't add additional handshake control.
-	if (getBituSubstring("transparent:", &bool_temp, cmd)) {
+	if (getUintFromString("transparent:", bool_temp, cmd)) {
 		if (bool_temp==1) transparent=true;
 		else transparent=false;
 	}
 	// telnet: interpret telnet commands.
-	if (getBituSubstring("telnet:", &bool_temp, cmd)) {
+	if (getUintFromString("telnet:", bool_temp, cmd)) {
 		if (bool_temp==1) {
 			transparent=true;
 			telnet=true;
 		}
 	}
 	// rxdelay: How many milliseconds to wait before causing an
 	// overflow when the application is unresponsive.
-	if (getBituSubstring("rxdelay:", &rx_retry_max, cmd)) {
+	if (getUintFromString("rxdelay:", rx_retry_max, cmd)) {
 		if (!(rx_retry_max<=10000)) {
 			rx_retry_max=50;
 		}
 	}
 	// txdelay: How many milliseconds to wait before sending data.
 	// This reduces network overhead quite a lot.
-	if (getBituSubstring("txdelay:", &tx_gather, cmd)) {
+	if (getUintFromString("txdelay:", tx_gather, cmd)) {
```

Comment:
We've renamed `getBitu...`  `getUint...` and now take in a reference to a `uint32_t` which we populate (instead of a pointer to a `Bitu`).